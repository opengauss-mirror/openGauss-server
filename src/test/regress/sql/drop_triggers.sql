CREATE SCHEMA drop_triggers_test;

SET search_path TO drop_triggers_test;

CREATE TABLE IF NOT EXISTS yx_channel (
    yx_id bigint NOT NULL,
    channel tinyint  NOT NULL,
    raw_data tinyint DEFAULT NULL,
    q bigint DEFAULT NULL,
    ld_alias varchar(64) DEFAULT NULL,
    order_no smallint DEFAULT '-1',
    gin bigint DEFAULT NULL,
    source_mode tinyint DEFAULT NULL,
    change_time bigint DEFAULT '0',
    PRIMARY KEY (yx_id,channel)
);

CREATE or replace FUNCTION "ti_yx_channel_trigger_function"() RETURNS "trigger"
    LANGUAGE "plpgsql" NOT SHIPPABLE
AS $$
DECLARE
    num1 INTEGER;
BEGIN
    RETURN NEW;
END$$;

CREATE TRIGGER "ti_yx_channel" BEFORE INSERT ON "yx_channel" FOR EACH ROW EXECUTE PROCEDURE "ti_yx_channel_trigger_function"();

-- ok
drop table yx_channel;
drop TRIGGER IF EXISTS  ti_yx_channel on yx_channel;


CREATE TABLE IF NOT EXISTS yx_channel (
    yx_id bigint NOT NULL,
    channel tinyint  NOT NULL,
    raw_data tinyint DEFAULT NULL,
    q bigint DEFAULT NULL,
    ld_alias varchar(64) DEFAULT NULL,
    order_no smallint DEFAULT '-1',
    gin bigint DEFAULT NULL,
    source_mode tinyint DEFAULT NULL,
    change_time bigint DEFAULT '0',
    PRIMARY KEY (yx_id,channel)
);

CREATE TRIGGER "ti_yx_channel" BEFORE INSERT ON "yx_channel" FOR EACH ROW EXECUTE PROCEDURE "ti_yx_channel_trigger_function"();

-- not ok
drop table yx_channel;
drop TRIGGER  ti_yx_channel on yx_channel;
drop TRIGGER IF EXISTS  ti_yx_channel on yx_channel;



CREATE TABLE IF NOT EXISTS yx_channel (
    yx_id bigint NOT NULL,
    channel tinyint  NOT NULL,
    raw_data tinyint DEFAULT NULL,
    q bigint DEFAULT NULL,
    ld_alias varchar(64) DEFAULT NULL,
    order_no smallint DEFAULT '-1',
    gin bigint DEFAULT NULL,
    source_mode tinyint DEFAULT NULL,
    change_time bigint DEFAULT '0',
    PRIMARY KEY (yx_id,channel)
);

CREATE TRIGGER "ti_yx_channel" BEFORE INSERT ON "yx_channel" FOR EACH ROW EXECUTE PROCEDURE "ti_yx_channel_trigger_function"();

drop TRIGGER ti_yx_channel on yx_channel;

drop TRIGGER IF EXISTS ti_yx_channel on yx_channel;

DROP SCHEMA drop_triggers_test CASCADE

