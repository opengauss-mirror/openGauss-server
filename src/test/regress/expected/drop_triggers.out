CREATE SCHEMA drop_triggers_test;
SET search_path TO drop_triggers_test;
CREATE TABLE IF NOT EXISTS yx_channel (
    yx_id bigint NOT NULL,
    channel tinyint  NOT NULL,
    raw_data tinyint DEFAULT NULL,
    q bigint DEFAULT NULL,
    ld_alias varchar(64) DEFAULT NULL,
    order_no smallint DEFAULT '-1',
    gin bigint DEFAULT NULL,
    source_mode tinyint DEFAULT NULL,
    change_time bigint DEFAULT '0',
    PRIMARY KEY (yx_id,channel)
);
NOTICE:  CREATE TABLE / PRIMARY KEY will create implicit index "yx_channel_pkey" for table "yx_channel"
CREATE or replace FUNCTION "ti_yx_channel_trigger_function"() RETURNS "trigger"
    LANGUAGE "plpgsql" NOT SHIPPABLE
AS $$
DECLARE
    num1 INTEGER;
BEGIN
    RETURN NEW;
END$$;
CREATE TRIGGER "ti_yx_channel" BEFORE INSERT ON "yx_channel" FOR EACH ROW EXECUTE PROCEDURE "ti_yx_channel_trigger_function"();
-- ok
drop table yx_channel;
drop TRIGGER IF EXISTS  ti_yx_channel on yx_channel;
NOTICE:  trigger "yx_channel.ti_yx_channel" for table "yx_channel" does not exist, skipping
CREATE TABLE IF NOT EXISTS yx_channel (
    yx_id bigint NOT NULL,
    channel tinyint  NOT NULL,
    raw_data tinyint DEFAULT NULL,
    q bigint DEFAULT NULL,
    ld_alias varchar(64) DEFAULT NULL,
    order_no smallint DEFAULT '-1',
    gin bigint DEFAULT NULL,
    source_mode tinyint DEFAULT NULL,
    change_time bigint DEFAULT '0',
    PRIMARY KEY (yx_id,channel)
);
NOTICE:  CREATE TABLE / PRIMARY KEY will create implicit index "yx_channel_pkey" for table "yx_channel"
CREATE TRIGGER "ti_yx_channel" BEFORE INSERT ON "yx_channel" FOR EACH ROW EXECUTE PROCEDURE "ti_yx_channel_trigger_function"();
-- not ok
drop table yx_channel;
drop TRIGGER  ti_yx_channel on yx_channel;
ERROR:  relation "yx_channel" does not exist
drop TRIGGER IF EXISTS  ti_yx_channel on yx_channel;
NOTICE:  trigger "yx_channel.ti_yx_channel" for table "yx_channel" does not exist, skipping
CREATE TABLE IF NOT EXISTS yx_channel (
    yx_id bigint NOT NULL,
    channel tinyint  NOT NULL,
    raw_data tinyint DEFAULT NULL,
    q bigint DEFAULT NULL,
    ld_alias varchar(64) DEFAULT NULL,
    order_no smallint DEFAULT '-1',
    gin bigint DEFAULT NULL,
    source_mode tinyint DEFAULT NULL,
    change_time bigint DEFAULT '0',
    PRIMARY KEY (yx_id,channel)
);
NOTICE:  CREATE TABLE / PRIMARY KEY will create implicit index "yx_channel_pkey" for table "yx_channel"
CREATE TRIGGER "ti_yx_channel" BEFORE INSERT ON "yx_channel" FOR EACH ROW EXECUTE PROCEDURE "ti_yx_channel_trigger_function"();
drop TRIGGER ti_yx_channel on yx_channel;
drop TRIGGER IF EXISTS ti_yx_channel on yx_channel;
NOTICE:  trigger "yx_channel.ti_yx_channel" for table "yx_channel" does not exist, skipping
DROP SCHEMA drop_triggers_test CASCADE
NOTICE:  drop cascades to 2 other objects
DETAIL:  drop cascades to function ti_yx_channel_trigger_function()
drop cascades to table yx_channel
